### PR TITLE
Code split demo changes and various other things

### PR DESCRIPTION
**Pull request by bryans99**
Note that main bundle size is still quite large. This is expected to be reduced with planned changes to looker components and extension sdk react

1. Major scenes are now loaded using react lazy
2. Optimization added to build
3. Upgraded looker components
4. Moved packages needed for development to devDependencies section of package.json
5. Added bundle analysis
6. Replaced custom extension button with looker component button outline
7. Replaced custom layout with looker component layout
8. Replaced custom message component with looker component textarea

NOTE THIS WILL BE A LONG LIVED PR. IT WILL BE NOT BE MERGED UNTIL A CORRESPONDING CHANGE IN HELLTOOL HAS BE RELEASED.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# VERY IMPORTANT NOTE - PLEASE READ
+
+This version and versions going forward demonstrate the use of code splitting. You must be on Looker version 7.21 or above for code splitting to work correctly.
+
+To remove code splitting you will need to modify the `KitchenSink.tsx`. Instructions are provided in this file to show how to do remove code splitting.
```
Comment (whscullin) [inline README.md:5]: ~do~

File: README.md [hunk 2]
```diff
@@ -342,11 +348,11 @@ const auth0Signin = async () => {
 
 ## Deployment
 
-The process above requires your local development server to be running to load the extension code. To allow other people to use the extension, we can build the JavaScript file and include it in the project directly.
+The process above requires your local development server to be running to load the extension code. To allow other people to use the extension, a production build of the extension needs to be run. As the kitchensink uses code splitting to reduce the size of the initially loaded bundle, multiple javascript files are generated.
```
Comment (whscullin) [inline README.md:351]: You have been CamelCasing JavaScript previously.

**Review by whscullin — COMMENTED** [review]
Generally LGTM

**Review by jkaster — COMMENTED** [review]
Very useful to see the shape of what's coming. Thanks for adding me to the PR review.

File: README.md (hunk 1 above)
Comment (bryans99) [reply README.md]: Fixed. Thanks for pointing out

File: README.md (hunk 2 above)
Comment (bryans99) [reply README.md]: Fixed. Thanks for pointing out

**Comment by bryans99** [comment]
Added code splitting example to the README

**Review by jkaster — APPROVED** [review]
LGTM

**Comment by bryans99** [comment]
Closed in favor of https://github.com/looker-open-source/extension-template-kitchensink/pull/49

**Pull request closed without merging**